File: tests/fixtures/languages/javascript/animals.js
import fs from "fs";
import { readFile } from "node:fs";

class Animal {
  speak() { return "..."; }
}

class Dog extends Animal {
  name = "rex";

  constructor(name) {
    super();
    this.name = name;
  }

  bark() {
    return this.speak();
  }
}

function makeDog(name) {
  const d = new Dog(name);
  return d.bark();
}

File: tests/fixtures/languages/python/animals.py
import os
from typing import List


class Animal:
    pass


class Dog(Animal):
    """A dog."""

    species: str = "canine"

    def __init__(self, name):
        self.name = name

    def bark(self) -> str:
        return self.speak()

    def speak(self):
        return "woof"


@app.route("/dogs", methods=["GET"])
def list_dogs():
    d = Dog("rex")
    return d.bark()
